.gitmodules

```
@@ -0,0 +1,9 @@
+[submodule "deps/json"]
+	path = deps/json
+	url = https://github.com/nlohmann/json.git
+[submodule "deps/fmt"]
+	path = deps/fmt
+	url = https://github.com/fmtlib/fmt.git
```

For simplicity I'd use the same name for each module and its corresponding `.cmake` file. Naming irregularities always make scripting things more annoying.

```suggestion
[submodule "deps/nlohmann-json"]
	path = deps/nlohmann-json
	url = https://github.com/nlohmann/json.git
[submodule "deps/fmtlib"]
	path = deps/fmtlib
	url = https://github.com/fmtlib/fmt.git
```

Also, more than one C++ JSON lib exists, and it's not even impossible that we'll switch again. Each one should get a unique dir name.